Registry/AvailableModule.cs

```diff
@@ -58,26 +59,39 @@ public CkanModule Latest(KSPVersion ksp_version = null)
             // Sort most recent versions first.            
             available_versions.Reverse();
 
-            if (ksp_version == null)
+            if (ksp_version == null && relationship == null)
             {
-                CkanModule module = module_version[available_versions.First()];
+                module = module_version[available_versions.First()];
 
                 log.DebugFormat("No KSP version restriction, {0} is most recent", module);
                 return module;
             }
+            if (relationship == null)
+            {
+                // Time to check if there's anything that we can satisfy.
+                var version =
+                    available_versions.FirstOrDefault(v => module_version[v].IsCompatibleKSP(ksp_version));
+                if (version != null)
+                    return module_version[version];
 
-            // Time to check if there's anything that we can satisfy.
+                log.DebugFormat("No version of {0} is compatible with KSP {1}",
+                    module_version[available_versions[0]].identifier, ksp_version);
 
-            foreach (Version v in available_versions.Where(v => module_version[v].IsCompatibleKSP(ksp_version)))
+                return null;
+            }
+            if (ksp_version == null)
             {
-                return module_version[v];
+                var version = available_versions.FirstOrDefault(relationship.version_within_bounds);
```

Comment:
So if I understand this correctly, a relationship allows us to specify a list of versions that the module can be? Such as "greater than 1.337, but less than 2.000?".